src/webhooks/helpers.ts

```diff
@@ -0,0 +1,60 @@
+// const { webcrypto, createHmac } = require("node:crypto");
+const { webcrypto } = require("node:crypto");
+const { subtle } = webcrypto;
+
+const ALGORITHM = { name: "HMAC", hash: "SHA-256" };
+
+const createHS256Signature = async ({ signingSecret, data }: { signingSecret: string; data: string }) => {
+    const enc = new TextEncoder();
+
+    let key = await subtle.importKey("raw", enc.encode(signingSecret), ALGORITHM, false, ["sign"]);
+    let signatureBuffer = await subtle.sign(ALGORITHM.name, key, enc.encode(data));
+    let signature = Buffer.from(signatureBuffer).toString("base64");
+
+    // Useful to check signature creation
+    // const hash = createHmac("sha256", signingSecret).update(data).digest("base64");
+    // console.log({ signature, hash });
+
+    return signature;
+};
+
+export type HookdeckVerifyArguments = {
+    signature: string;
+    secondarySignture?: string;
+    rawBody: string;
+    signingSecret: string;
+};
+
+export type HookdeckVerificationResult = {
+    isValidSignature: boolean;
+};
+
+/**
+ * Verify the Hookdeck webhook signature.
+ *
+ * @example
+ *     const result = await verifyWebhookSignature({
+ *         signature: "kKb0yljhY9tBo7oihOMTvRayKCUpCNVujkoTMNoGdGM=",
+ *         rawBody: `{"key1":"value1", "key2":"value2"}`,
+ *         signingSecret: "5fthvdmj8gtkzdv93mwkdpbupgwgfu4fu8xf4rro2oufn6qlhc",
+ *     })
+ *
+ *     if(!result.isValidSignature) {
+ *         // Reject the webhook payload
+ *     }
+ *     else {
+ *         // Proceed with the webhook payload
+ *     }
+ */
+export const verifyWebhookSignature = async ({
+    signature,
+    rawBody,
+    signingSecret,
+    secondarySignture,
+}: HookdeckVerifyArguments): Promise<HookdeckVerificationResult> => {
+    const signatureCheck = await createHS256Signature({ signingSecret, data: rawBody });
+
+    return {
+        isValidSignature: signatureCheck === signature || signatureCheck === secondarySignture,
```

Comment:
Need to decide how to handle `x-hookdeck-verified`, a flag to the method maybe?

Reply:
What do you think of the DX of making the `verifyWebhookSignature` function require a key/value parameter of `headers`? That way the helper can extract the required headers and the calling of the function is more about telling it how to behave.

```
const result =
  await verifyWebhookSignature(
    {
      headers,
      config: {
        checkSourceVerification: true, // true: check for x-hookdeck-verified and fails if it's false
      }
    }
  );
```

Reply:
Here's a full example with the two options:

```ts
import { NextResponse } from "next/server";
import { verifyWebhookSignature } from "@hookdeck/sdk/webhooks/helpers";

export async function POST(request: Request) {

  // Option 1: explicitly pass in headers and other params
  const verificationResult = await verifyWebhookSignature({
    signature: request.headers.get("x-hookdeck-signature")!,
    secondarySignature: request.headers.get("x-hookdeck-signature-2"),
    sourceVerified: request.headers.get("x-hookdeck-verified"),
    rawBody: await request.text(),
    signingSecret: process.env.HOOKDECK_SIGNING_SECRET!,
  });

  // Option 2: developer needs to create header key/value pair
  // then pass in headers
  const headers: { [key: string]: string } = {};
  request.headers.forEach((value, key) => {
    headers[key] = value;
  });

  const verificationResult = await verifyWebhookSignature({
    headers,
    rawBody: await request.text(),
    signingSecret: process.env.HOOKDECK_SIGNING_SECRET!,
    config: { checkSourceVerification: true },
  });

  if (!verificationResult.isValidSignature) {
    return new NextResponse(
      JSON.stringify({ error: "Invalid Hookdeck signature" }),
      { status: 401 },
    );
  }
  const body = await request.json();
  console.log(body);
  return NextResponse.json({ received: true });
}
```

I think option 2 is a little more work to create the `headers` key-value pair. However, it removes the chances of people making errors when getting the header params and hides more of the implementation details.

Reply:
Yeah I believe #2 is the most practical

> I think option 2 is a little more work to create the headers key-value pair. 

Not sure what you mean by that!

Reply:
> > I think option 2 is a little more work to create the headers key-value pair.
> 
> Not sure what you mean by that!

Getting the headers is a little more work for the developer using the SDK. Not more work to be done within the SDK.